web-ui/src/app/components/latest-blocks/latest-blocks.component.ts

```diff
@@ -70,9 +72,30 @@ export class LatestBlocksComponent implements OnInit, OnDestroy {
   }
 
   private onBlockRetrieved(response: Block[]) {
+    this.countNewestBlocks(response);
+    this.latestBlock = response[0];
     this.blocks = response;
   }
 
+  private countNewestBlocks(newBlocks: Block[]) {
+    if (!this.blocks) {
+      return;
+    }
+
+    const lastestBlockHash: string = this.latestBlock.hash;
+    let totalNewBlocks = 0;
+
+    for (let i = 0; i < newBlocks.length; i++) {
```

Comment:
A loop is error prone, use collection transformation functions instead, like `map` and `filter`, if you use the latest block height, it is simple to filter by the blocks having a higher height.